### DISCUSSION TITLE
Slow loading of pages with a lot of Youtube entries

### DISCUSSION THREAD

**Discussion by Alwaysin** [FreshRSS/FreshRSS#6431] [General]
Not sure if this is a bug so I'm starting a discussion instead. When displaying a Youtube feed, the page takes way more time to finish rendering completly than it takes when there are no Youtube feed entries. I've noticed a lot of "grey.gif" images are loaded when it does so (as seen by the "Network" tab on the browser's developer tools), maybe a fallback mechanism of some sort?

This seems to be the relevant part of the code loading the gif:
https://github.com/FreshRSS/FreshRSS/blob/fd7157e40ca32fb1bcda51f1aaf484cbb7988bdd/lib/lib_rss.php#L609-L619

There really seems to be a strange behaviour somewhere, because when I uncheck "Use lazy load mode to load pictures" in reading configuration and I go back on a Youtube feed, it loads quickly BUT there are so many requests going on after a little while initiated by `youtube.com` domain and others (doubleclick trying to load ads, etc) that it looks like every feed entry is entirely loaded as if I was clicking and opening every link at the same time.

I'm maybe not very clear in my explanation, please try by yourself with a Youtube RSS link and the developer tools

**Comment by Frenzie** [comment]
I'm not quite understanding what the problem is? A YouTube iframe doesn't just load an entirely different website, but one that's significantly heavier to boot. I'd be quite surprised if it didn't take longer to fully finish loading. But everything is fully usable long before then.

**Reply by Alwaysin** [reply]
All this is happening just by loading `https://domain/i/?a=normal&get=f_86`, not clicking on any entry. Then I guess it is supposed to load all the content of every entries as I understand and we can't do anything about it.
It even seems to be looping, e.g. on a page with 83 entries, after 3 minutes (it was still continuing):
![chrome_tBKn2ZuJUm](https://github.com/FreshRSS/FreshRSS/assets/7189936/26aa9063-3f5d-46bc-96fd-bc0a37d6b17e)
26k errors and 30k+ warning in console log ^^'

**Comment by math-GH** [comment]
Do you use an extension for Youtube?

I have subscribed to youtube feeds too but it displays just an image instead of an embedded video.
Or do you have subscribed to a feed, that has embedded youtube videos?

Could you share the link to the feed?

**Reply by Alwaysin** [reply]
Woops, newbie error. It is caused by https://github.com/FreshRSS/Extensions/tree/master/xExtension-YouTube

**Reply by math-GH** [reply]
Could you please create an issue ticket in the extension repository?

**Reply by Alwaysin** [reply]
Done. This discussion should be deleted.

**Reply by Alkarex** [reply]
https://github.com/FreshRSS/Extensions/issues/236
